### PR TITLE
Angle and Offset updates

### PR DESCRIPTION
**Pull request by swalker2m**
A couple of math updates:

* `Angle` gets unsigned milliarcsecond and arcsecond optics
* `Offset` gets a `distance` to compute angular separation between two offsets

### CODE REVIEW

**Review by rpiaggio — APPROVED** [review]
LGTM

File: modules/core/shared/src/main/scala/lucuma/core/math/Offset.scala [hunk 1]
```diff
@@ -51,6 +51,16 @@ final case class Offset(p: Offset.Component[Axis.P], q: Offset.Component[Axis.Q]
   def rotate(θ: Angle): Offset =
     Offset.rotateBy(θ)(this)
 
+  /**
+   * Angular separation between two offset positions.
+   */
+  def distance(o: Offset): Angle = {
```
Comment (cquiroz) [inline modules/core/shared/src/main/scala/lucuma/core/math/Offset.scala:57]: I guess this is for relatively small offsets?

Comment (swalker2m) [reply modules/core/shared/src/main/scala/lucuma/core/math/Offset.scala:57]: In practice yes, but it's angular separation not RA/Dec coordinates so the declination is not relevant.  It should work regardless of the separation.

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/Offset.scala:57]: ah ok cool!

**Pull request merged**